scripts/locust/service.py

```diff
@@ -40,120 +81,251 @@ def parallel_requests():
-                self._sleep(retry_after)
-                with self.client.get(refresh_url, catch_response=True, allow_redirects=False) as response:
-                    if response.status_code in [301, 302]:
-                        response.success()
-
-        @seq_task(3)
-        def stop(self):
-            self.interrupt()
+    @task
+    def projects_start_page(self):
+        # By default data browser only shows human data
+        self.browse_page('projects',
+                         {"genusSpecies": {"is": ["Homo sapiens"]}},
+                         size=15,
+                         sort='projectTitle',
+                         order='asc')
+
+    @task
+    def filter_mtx_files(self):
+        self.browse_page('projects',
+                         {"fileFormat": {"is": ["mtx"]}},
+                         **self.browser_search_params)
+
+
+class ManifestTaskSet(BrowserTaskSet, FileFetchTaskSet):
+    """
+    Filter files by organ part and download a BDBag-format manifest.
+    """
+
+    # Select islet of Langerhans since it's present in the develop deployment.
+    organ_part_filter = {"organPart": {"is": ["islet of Langerhans"]}}
+    manifest_file_format_filter = {
+        "fileFormat": {
+            "is": ["fastq.gz", "bai", "bam", "csv", "results", "txt"]
+        }
+    }
+
+    @task
+    def start_page(self):
+        self.browse_page('files', {}, size=15)
+
+    @task
+    def filter_organ_part(self):
+        self.browse_page('files',
+                         self.organ_part_filter,
+                         **self.browser_search_params)
+
+    @task
+    def download_manifest(self):
+        self.client.get(self.endpoint('/index/summary', filters=json.dumps(self.organ_part_filter)),
+                        name='/index/summary')
+        export_url = self.endpoint('/manifest/files',
+                                   filters=json.dumps({**self.organ_part_filter,
+                                                       **self.manifest_file_format_filter}),
+                                   format='terra.bdbag')
+        self.download_redirects(export_url, '/manifest/files')
+
+
+class RepositoryTaskSet(FileSelectionTaskSet, FileFetchTaskSet):
+    """
+    Select and download a file using both fetch and non-fetch approaches.
+    """
+
+    @task
+    def _choose_file(self):
+        super().choose_file({"fileFormat": {"is": ["fastq", "fastq.gz"]}})
+
+    def _download(self, path: str):
+        url = self.endpoint(path.format(file_uuid=self.file_uuid))
+        self.download_redirects(url, path)
+
+    @task
+    def file(self):
+        self._download('/repository/files/{file_uuid}')
+
+    @task
+    def fetch_file(self):
+        self._download('/fetch/repository/files/{file_uuid}')
+
+
+class GA4GHTaskSet(FileSelectionTaskSet):
+    """
+    Test DRS and DOS endpoints after selecting a file.
+    """
+
+    # Our DRS implementation only works for files in the DCP1 catalog
+    catalog = 'dcp1'
```

Comment:
This looks like its silently overriding the configured catalog. It should be skipped for catalogs other than those configured to use the DSSRepoPlugin. IOW, don't hardcode catalog names, if possible.